src/scripts/main.js

```diff
@@ -1,24 +1,39 @@
+/* eslint-disable no-shadow */
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, function handler(event) {
+      element.removeEventListener(eventName, handler);
+
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+    });
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const div = document.createElement('div');
+
+  div.classList.add('message');
+  div.textContent = message;
+  document.body.appendChild(div);
 };
 
-const loginField = document.getElementById('login');
-const passwordField = document.getElementById('password');
-const button = document.getElementById('submit');
+document.addEventListener('DOMContentLoaded', () => {
+  const loginField = document.getElementById('login');
+  const passwordField = document.getElementById('password');
+  const button = document.getElementById('submit');
 
-waitFor(loginField, 'click').then(printMessage);
-waitFor(passwordField, 'click').then(printMessage);
-waitFor(button, 'click').then(printMessage);
+  waitFor(loginField, 'click').then(printMessage);
+  waitFor(passwordField, 'click').then(printMessage);
+  waitFor(button, 'click').then(printMessage);
 
-waitFor(loginField, 'input').then(printMessage);
-waitFor(passwordField, 'input').then(printMessage);
+  waitFor(loginField, 'input').then(printMessage);
+  waitFor(passwordField, 'input').then(printMessage);
 
-waitFor(loginField, 'blur').then(printMessage);
-waitFor(passwordField, 'blur').then(printMessage);
-waitFor(button, 'blur').then(printMessage);
+  waitFor(loginField, 'blur').then(printMessage);
+  waitFor(passwordField, 'blur').then(printMessage);
+  waitFor(button, 'blur').then(printMessage);
```

Comment:
The 'blur' event is useful for detecting when an element loses focus. Ensure that the `printMessage` function is suitable for handling the messages generated by these events, as it might not be necessary to print messages for every blur event.